src/backend/account.rs

```diff
@@ -0,0 +1,32 @@
+use zbus::dbus_interface;
+use zvariant::DeserializeDict;
+
+use crate::{
+    desktop::{
+        account::UserInfo,
+        request::{Response, ResponseError},
+        HandleToken,
+    },
+    WindowIdentifierType,
+};
+
+#[derive(Debug, DeserializeDict, zvariant::Type)]
+#[zvariant(signature = "dict")]
+pub struct UserInfoOptions {
+    reason: Option<String>,
+}
+
+pub struct Account {}
+
+#[dbus_interface(name = "org.freedesktop.impl.portal.Account")]
+impl Account {
+    async fn get_user_information(
+        &self,
+        handle: HandleToken,
+        app_id: &str,
+        window_identifier: WindowIdentifierType,
+        options: UserInfoOptions,
+    ) -> Response<UserInfo> {
+        Response::Err(ResponseError::Cancelled)
```

Comment:
Is there a reason Response is like Result instead of modeling the apis around returning a request::Result<UserInfo> type?

Reply:
The main reason was the custom deserialize/serialize that are needed for Response;  we can always make that an internal implementation.